mycpp/gc_list.h

```diff
@@ -295,17 +297,15 @@ T List<T>::pop() {
 }
 
 // Used in osh/word_parse.py to remove from front
-// TODO: Don't accept an arbitrary index?
 template <typename T>
 T List<T>::pop(int i) {
   DCHECK(len_ > 0);
-  DCHECK(i == 0);  // only support popping the first item
 
-  T result = index_(0);
+  T result = index_(i);
   len_--;
 
   // Shift everything by one
-  memmove(slab_->items_, slab_->items_ + 1, len_ * sizeof(T));
+  memmove(slab_->items_ + i, slab_->items_ + (i + 1), len_ * sizeof(T));
```

Comment:
I _think_ this should be fine in general... Perhaps we could add a few extra tests to convince ourselves if unsure